File: src/main/java/com/github/caijh/commons/util/ProcessUtils.java
package com.github.caijh.commons.util;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.nio.charset.StandardCharsets;

public class ProcessUtils {

    private ProcessUtils() {
    }

    public static String readConsoleString(Process process) throws IOException {
        StringBuilder console = new StringBuilder();
        BufferedReader bufferIn = new BufferedReader(new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8));
        BufferedReader bufferedError = new BufferedReader(new InputStreamReader(process.getErrorStream(), StandardCharsets.UTF_8));

        String line;
        while ((line = bufferIn.readLine()) != null || (line = bufferedError.readLine()) != null) {
            console.append(line).append('\n');
        }
        return console.toString();
    }

}

File: src/main/kotlin/com/github/caijh/commons/util/process/ProcessUtils.kt
package com.github.caijh.commons.util.process

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.nio.charset.StandardCharsets

object ProcessUtils {
    @Throws(IOException::class)
    fun readConsoleString(process: Process): String {
        val console = StringBuilder()
        val bufferIn = BufferedReader(InputStreamReader(process.inputStream, StandardCharsets.UTF_8))
        val bufferedError = BufferedReader(InputStreamReader(process.errorStream, StandardCharsets.UTF_8))
        var line: String?
        while (bufferIn.readLine().also { line = it } != null || bufferedError.readLine().also { line = it } != null) {
            console.append(line).append('\n')
        }
        return console.toString()
    }
}
